packages/analytics/shippers/elastic_v3/server/src/server_shipper.ts

```diff
@@ -216,40 +215,29 @@ export class ElasticV3ServerShipper implements IShipper {
   }
 
   private setInternalSubscriber() {
-    // Check the status of the queues every 1 second.
+    // Create an emitter that emits when 10s have passed since the last time we sent the data
+    const atLeast10secondsSinceLastSent$ = interval(SECOND).pipe(
+      filter(() => Date.now() - this.lastBatchSent >= 10 * SECOND)
+    );
+
     merge(
-      interval(1000).pipe(takeUntil(this.shutdown$)),
-      // Using a promise because complete does not emit through the pipe.
-      from(firstValueFrom(this.shutdown$, { defaultValue: true }))
+      atLeast10secondsSinceLastSent$.pipe(takeUntil(this.shutdown$)),
+      // Attempt to send one last time on shutdown
+      this.shutdown$
     )
       .pipe(
-        // Only move ahead if it's opted-in and online.
-        filter(() => this.isOptedIn$.value === true && this.firstTimeOffline === null),
-
-        // Send the events now if (validations sorted from cheapest to most CPU expensive):
-        // - We are shutting down.
-        // - There are some events in the queue, and we didn't send anything in the last 10 minutes.
-        // - The last time we sent was more than 10 seconds ago and:
-        //   - We reached the minimum batch size of 10kB per request in our leaky bucket.
-        //   - The queue is full (meaning we'll never get to 10kB because the events are very small).
+        // Only move ahead if it's opted-in and online, and there are some events in the queue
         filter(
           () =>
-            (this.internalQueue.length > 0 &&
-              (this.shutdown$.isStopped || Date.now() - this.lastBatchSent >= 10 * MINUTE)) ||
-            (Date.now() - this.lastBatchSent >= 10 * SECOND &&
-              (this.internalQueue.length === MAX_NUMBER_OF_EVENTS_IN_INTERNAL_QUEUE ||
-                this.getQueueByteSize(this.internalQueue) >= 10 * KIB))
+            this.isOptedIn$.value === true &&
+            this.firstTimeOffline === null &&
+            this.internalQueue.length > 0
         ),
-
-        // Send the events:
-        // 1. Set lastBatchSent and retrieve the events to send (clearing the queue) in a synchronous operation to avoid race conditions.
+        // Retrieve the events to send (clearing the queue) in a synchronous operation to avoid race conditions.
         map(() => {
           this.lastBatchSent = Date.now();
           return this.getEventsToSend();
         }),
-        // 2. Skip empty buffers
-        filter((events) => events.length > 0),
```

Comment:
Forgive the naive question, but: Does removing this empty check open us up to more incidents of receiving empty bodies at the endpoint? 😬

Reply:
According to the current implementation, it should only move ahead if there's anything to send in the queue. So this check was redundant. That said, I'm happy to bring it back for the sake of completeness